### ISSUE TITLE
"Use Snap" in 3D editor is weird

### ISSUE THREAD

**Issue by donn-xx** [godotengine/godot#95229]
### Tested versions

v4.3.rc.custom_build [4359c28fe]

### System information

Godot v4.3.rc (4359c28fe) - Ubuntu 22.04.4 LTS 22.04 - X11 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 2060 (nvidia; 535.183.01) - Intel(R) Core(TM) i5-9400F CPU @ 2.90GHz (6 Threads)

### Issue description

It's hard to pin-down. I searched a lot for another issue like it, but found nothing. Hope it's unique.

Roughly:
1. Switch between scenes
2. Select an object
3. Grab one of the basis handles and start dragging it to move the object
4. It will either move smoothly (1 px at a time) or it will seem to dig its heels-in and refuse.
5. If it refuses, then **snap** is on—but this does not always sync with the visual pressed state of the **snap** button on the menu, which can appear off when it's on and vice-versa.
6. Toggling snap mostly fixes it, for a moment, but before long it gets confused again. Goto 1 :)

Thought: My project uses objects are all about 10cm cubed. Could it be the zoom size, or something?

### Steps to reproduce

It's a pure editor thing, afaict. No project required.

### Minimal reproduction project (MRP)

None

**Comment by theraot** [comment]
I'm not sure if it is the same as this:

- https://github.com/godotengine/godot/issues/90320

**Comment by donn-xx** [comment]
> I'm not sure if it is the same as this:
>     * ["Use snap (Y)" behavior is inconsistent #90320](https://github.com/godotengine/godot/issues/90320)

Yes, seems very similar, but I don't even need to use Ctrl+A to trigger it.

**Comment by Zennii** [comment]
I can seem to get this to happen when saving (ctrl+S) which is common. So that could also be what is being experienced here.

**Comment by donn-xx** [comment]
> I can seem to get this to happen when saving (ctrl+S) which is common. So that could also be what is being experienced here.

I tried this in various places and can't reproduce.
